core/src/main/java/org/apache/gravitino/metalake/MetalakeManager.java

```diff
@@ -103,10 +132,12 @@ public static void checkMetalake(NameIdentifier ident, EntityStore store)
   public static boolean metalakeInUse(EntityStore store, NameIdentifier ident)
       throws NoSuchMetalakeException {
     try {
-      BaseMetalake metalake = store.get(ident, EntityType.METALAKE, BaseMetalake.class);
+      BaseMetalake metalake = METALAKE_CACHE.getIfPresent(ident);
+      if (metalake == null) {
+        metalake = store.get(ident, EntityType.METALAKE, BaseMetalake.class);
+      }
```

Comment:
why not cache the result after getting it from the store?

Reply:
Please see https://github.com/apache/gravitino/pull/6569#discussion_r1974969120

Reply:
after the user alters the metalake, then load schema/table directly without list/get the metalake, the cache never be hitting, right?

Reply:
Yeah. alter a metalake is not operated frequently, so I guess it's acceptable